client/state/jetpack-connect/schema.js

```diff
@@ -2,7 +2,7 @@ export const jetpackConnectSessionsSchema = {
 	type: 'object',
 	additionalProperties: false,
 	patternProperties: {
-		'^[a-z0-9\.\-\:\/]+$': {
+		'^[a-z0-9\.\:\/\-]+$': {
```

Comment:
This is still a little too simple if you're matching a url. For example can this handle query parameters or anchors?

I would suggest adding in a more complete regex, or removing the pattern properties requirement here, since I assume this is coming from an endpoint
```javascript
^((http[s]?):\/)?\/?([^:\/\s]+)((\/\w+)*\/)([\w\-\.]+[^#?\s]+)(.*)?(#[\w\-]+)?$
```